server/app/views/ViewUtils.java

```diff
@@ -232,4 +234,67 @@ public PTag renderEditOnText(String prefix, Optional<Instant> time) {
   public static SpanTag requiredQuestionIndicator() {
     return span(rawHtml("&nbsp;*")).withClasses("text-red-600", "font-semibold");
   }
+
+  public static ButtonTag makeToggleButton(
+      String fieldName,
+      boolean enabled,
+      Optional<String> text,
+      Optional<String> instanceClassSuffix) {
+    String toggleInstanceClass = instanceClassSuffix.map((v) -> "cf-toggle-" + v).orElse("");
+    String toggleInstanceInputClass =
+        instanceClassSuffix.map((v) -> "cf-toggle-hidden-input-" + v).orElse("");
+    String toggleInstanceBackgroundClass =
+        instanceClassSuffix.map((v) -> "cf-toggle-background-" + v).orElse("");
+    String toggleInstanceNubClass = instanceClassSuffix.map((v) -> "cf-toggle-nub-" + v).orElse("");
+    ButtonTag button =
+        TagCreator.button()
+            .withClasses(
+                "cf-toggle",
+                "flex",
+                "px-0",
+                "gap-2",
+                "items-center",
+                "text-black",
+                "font-normal",
+                "bg-transparent",
+                "rounded-full",
+                StyleUtils.hover("bg-transparent"),
+                toggleInstanceClass)
+            .withType("button")
+            .with(
+                input()
+                    .isHidden()
+                    .withName(fieldName)
+                    .withClasses("cf-toggle-hidden-input", toggleInstanceInputClass)
+                    .withValue(Boolean.valueOf(enabled).toString()))
+            .with(
+                div()
+                    .withClasses("relative")
+                    .with(
+                        div()
+                            .withClasses(
+                                enabled ? "bg-blue-600" : "bg-gray-600",
+                                "w-14",
+                                "h-8",
+                                "rounded-full",
+                                "toggle",
+                                "cf-toggle-background",
+                                toggleInstanceBackgroundClass))
+                    .with(
+                        div()
+                            .withClasses(
+                                "absolute",
+                                "bg-white",
+                                enabled ? "right-1" : "left-1",
+                                "top-1",
+                                "w-6",
+                                "h-6",
+                                "rounded-full",
+                                "cf-toggle-nub",
+                                toggleInstanceNubClass)));
+    if (text.isPresent()) {
+      button.withText("Set as a universal question");
+    }
```

Comment:
issue (blocking): this text should be added somewhere else since this method is generic